src/eos-shard-writer-v2.c

```diff
@@ -143,13 +142,30 @@ constant_pool_write (struct constant_pool *cpool, int fd, off_t offset)
   }
 }
 
+static guint
+csum_hash (gconstpointer v)
+{
+  const signed char *p, *e;
+  guint32 h = 5381;
+  for (p = v, e = v+0x20; p < e; p++)
```

Comment:
I'll add a comment about it being g_str_hash.

gcc doesn't like the v+0x20 form because we're comparing a signed char* to a void*

I could use g_direct_hash but it just uses the first four bytes arbitrarily as a hash which isn't necessarily great...